byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/destinations")
+public class DestinationsServlet extends HttpServlet {
+
+    public static class UserLocations{ 
+        String start;
+        ArrayList<String> destinations;
+
+        public UserLocations(){
+            this.start = "";
+            this.destinations= new ArrayList<String>();
+        }
+    }
+
+  private final UserLocations places = new UserLocations();
```

Comment:
Okay, we plan on switching to datastore instead. Would that fix the issue?

Reply:
Yeah, datastore is appropriate for storing this kind of data.

If you intend to leave with fixed data for testing purposes (where you expect to have just one server instance, so the data will be shared sort of the way you expect), you should still apply synchronization so that you don't run into surprise concurrent access bugs.

You can add a [`@GuardedBy("places")`](https://errorprone.info/bugpattern/GuardedBy) annotation, and then in places where you need to modify or read the value of `places` wrap in `synchronized(places) { ... }`. Alternatively, you could use `@GuardedBy("this")` and just declare `doGet` and `doPost` as `synchronized doGet`. Note that either way this would be a pretty inefficient and a bad idea to do in production code, because it means that every request has to wait for all other requests using the same variable to be completed before it can take its turn, like a highway with only one lane open.

Reply:
Okay, I implemented datastore